### ISSUE TITLE
NormalMortarMechanicalContact with LowerDBlockFromSidesetGenerator causes logic error

### ISSUE THREAD

**Issue by shaneongit** [idaholab/moose#14082]
## Bug Description
NormalMortarMechanicalContact with LowerDBlockFromSidesetGenerator causes a subscript out-of-bounds at NormalMortarMechanicalContact.C:52 on the master side of the interface. This is the error message:

```
Time Step 1, time = 1, dt = 1
/opt/moose/gcc-7.3.1/include/c++/7.3.1/debug/vector:424:
Error: attempt to subscript container with out-of-bounds index 1, but 
container only holds 1 elements.

Objects involved in the operation:
    sequence "this" @ 0x0x2242fa0 {
      type = std::__debug::vector<double, std::allocator<double> >;
    }
Aborted (core dumped)
```

## Impact
The combination of NormalMortarMechanicalContact and LowerDBlockFromSidesetGenerator are needed to modernize the tests at modules/contact/test/tests/mechanical-small-problem as well as to enable creation of an Action that provides the objects needed for mortar mechanical contact: #14049.

## Steps to Reproduce
Run contact app with this input file:
```
[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  [./simple_mesh]
    type = FileMeshGenerator
    file = mesh.e
  [../]
  [./master]
    type = LowerDBlockFromSidesetGenerator
    input = simple_mesh
    sidesets = '2'
    new_block_id = '20'
  [../]
  [./slave]
    type = LowerDBlockFromSidesetGenerator
    input = simple_mesh
    sidesets = '1'
    new_block_id = '10'
  [../]
[]

[Constraints]
  [./lm]
    type = NormalNodalLMMechanicalContact
    slave = 1
    master = 2
    variable = lambda
    master_variable = disp_x
    disp_y = disp_y
  [../]
  [x]
    type = NormalMortarMechanicalContact
    master_boundary = '2'
    slave_boundary = '1'
    master_subdomain = 20
    slave_subdomain = 10
    variable = lambda
    slave_variable = disp_x
    component = x
    use_displaced_mesh = true
    compute_lm_residuals = false
  []
[]

[Variables]
  [./disp_x]
    block = '1 2'
  [../]
  [./disp_y]
    block = '1 2'
  [../]
  [./lambda]
    block = 10
  [../]
[]

[BCs]
  [./left_x]
    type = DirichletBC
    variable = disp_x
    boundary = 'outside_left'
    value = 0.0
  [../]
  [./left_y]
    type = DirichletBC
    variable = disp_y
    boundary = 'outside_left'
    value = 0.0
  [../]
  [./right_x]
    type = FunctionPresetBC
    variable = disp_x
    boundary = 'outside_right'
    function = '-5e-3 * t'
  [../]
  [./right_y]
    type = FunctionPresetBC
    variable = disp_y
    boundary = 'outside_right'
    function = 0
  [../]
[]

[Kernels]
  [disp_x]
    type = Diffusion
    variable = disp_x
    block = '1 2'
  []
  [disp_y]
    type = Diffusion
    variable = disp_y
    block = '1 2'
  []
[]

[Debug]
  show_var_residual_norms = 1
[]

[Preconditioning]
  [./smp]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  solve_type = PJFNK
  type = Transient
  num_steps = 10
  dt = 1
  dtmin = 1
  petsc_options_iname = '-pc_type -snes_linesearch_type -pc_factor_shift_type -pc_factor_shift_amount'
  petsc_options_value = 'lu       basic                 NONZERO               1e-15'
[]

[Outputs]
  exodus = true
[]
```

**Comment by lindsayad** [comment]
@shaneongit Can you also attach or email me the mesh file?

**Comment by shaneongit** [comment]
It's the mesh at modules/contact/test/tests/mechanical-small-problem/mesh.e. This problem is a copy of frictionless-nodal-lm-mortar-disp.i, but without the constraint in y (I just removed y to make stepping in the debugger a bit easier/quicker).

**Comment by lindsayad** [comment]
So if you change `Mesh/slave/input=simple_mesh` to `Mesh/slave/input=master` then this input file will run. It's a little curious to me that we don't have an error check for this (I actually thought that we did), but it needs to be obvious what the final mesh should be. As originally provided, the final mesh could be that generated after running `slave` or `master`; either way that final mesh wouldn't include the effects of both `slave` and `master`.

#14061 updates the contact tests to use mesh generators. I didn't check to see whether your WIP PR had also done that...

**Comment by lindsayad** [comment]
I'm closing this in favor of #14058 (which will be closed when #14060 goes in)
